core/common/src/main/java/alluxio/metrics/MetricsSystem.java

```diff
@@ -778,8 +778,17 @@ private static synchronized List<alluxio.grpc.Metric> reportMetrics(InstanceType
               entry.getKey(), gauge.getValue().getClass().getSimpleName());
           continue;
         }
-        rpcMetrics.add(Metric.from(entry.getKey(),
-            ((Number) gauge.getValue()).longValue(), MetricType.GAUGE).toProto());
+        // As the value is a number, it can also be reported to master as "COUNTER" type.
+        // Then it can also be aggregated in the same way as "COUNTER" at master side.
+        long value = ((Number) gauge.getValue()).longValue();
+        Long prev = LAST_REPORTED_METRICS.replace(entry.getKey(), value);
+        if (prev == null) {
+          LAST_REPORTED_METRICS.put(entry.getKey(), value);
+        }
+        double diff = prev != null ? value - prev : value;
+        if (diff != 0) { // Only report non-zero counter values
+          rpcMetrics.add(Metric.from(entry.getKey(), diff, MetricType.COUNTER).toProto());
```

Comment:
Please add a comment that explains why a gauge type metric needs to be converted to a counter when it's sent to metrics master for cluster aggregation.

https://github.com/Alluxio/alluxio/blob/a4ecde4b26c9661931bc860ca708ce8ec280ca39/core/server/master/src/main/java/alluxio/master/metrics/MetricsStore.java#L129

Reply:
> Please add a comment that explains why a gauge type metric needs to be converted to a counter

Added.